nixos/modules/system/boot/loader/systemd-boot/systemd-boot-builder.py

```diff
@@ -244,27 +258,29 @@ def main() -> None:
         subprocess.check_call(["@systemd@/bin/bootctl", "--path=@efiSysMountPoint@"] + flags + ["install"])
     else:
         # Update bootloader to latest if needed
-        systemd_version = subprocess.check_output(["@systemd@/bin/bootctl", "--version"], universal_newlines=True).split()[2]
-        sdboot_status = subprocess.check_output(["@systemd@/bin/bootctl", "--path=@efiSysMountPoint@", "status"], universal_newlines=True)
+        systemd = subprocess.check_output(["@systemd@/bin/bootctl", "--version"], universal_newlines=True).split()[2]
+        sdboot = subprocess.check_output(["@systemd@/bin/bootctl", "--path=@efiSysMountPoint@", "status"], universal_newlines=True)
 
         # See status_binaries() in systemd bootctl.c for code which generates this
-        m = re.search("^\W+File:.*/EFI/(BOOT|systemd)/.*\.efi \(systemd-boot ([\d.]+[^)]*)\)$",
-                      sdboot_status, re.IGNORECASE | re.MULTILINE)
+        sdboot_version_match = re.search("^\W+File:.*/EFI/(BOOT|systemd)/.*\.efi \(systemd-boot ([\d.]+[^)]*)\)$",
+                      sdboot, re.IGNORECASE | re.MULTILINE)
 
-        needs_install = False
+        systemd_version_match = re.search("^\((.*)\)$", systemd)
 
-        if m is None:
-            print("could not find any previously installed systemd-boot, installing.")
-            # Let systemd-boot attempt an installation if a previous one wasn't found
-            needs_install = True
-        else:
-            sdboot_version = f'({m.group(2)})'
-            if systemd_version != sdboot_version:
-                print("updating systemd-boot from %s to %s" % (sdboot_version, systemd_version))
-                needs_install = True
+        if sdboot_version_match is None:
+            raise Exception("could not find any previously installed systemd-boot")
+
+        if systemd_version_match is None:
+            raise Exception("could not determine systemd-boot version")
```

Comment:
This is unreachable code.

Reply:
I don't see why, per

> https://docs.python.org/3/library/re.html#re.search

`re.search` may return `None`, and `systemd_version_match` is

```
systemd_version_match = re.search("^\((.*)\)$", systemd)
```

Reply:
nvm sdboot_version_match != systemd_version_match